lib/linux/sys/proctable.rb

```diff
@@ -169,6 +171,10 @@ def self.ps(pid=nil)
         # Get control groups to which the process belongs
         struct.cgroup = IO.readlines("/proc/#{file}/cgroup").map { |l| CgroupEntry.new(l) } rescue []
 
+        # Read smaps, return a parseable empty string if we don't have permissions
+        smaps_contents = IO.read("/proc/#{file}/smaps") rescue ""
```

Comment:
Does it make sense to report 0 for vss, rss, uss, etc. if it's not readable?  That's what happening now: 

```
Sys::ProcTable.ps(1)
=> #<struct Struct::ProcTableStruct cmdline="/sbin/init", cwd=nil, environ={}, exe=nil, fd={}, root=nil, pid=1, comm="init", state="S", ppid=0, pgrp=1, session=1, tty_nr=0, tpgid=-1, flags=4202752, minflt=2809, cminflt=7384315, majflt=9, cmajflt=1034, utime=2, stime=137, cutime=6494, cstime=2250, priority=20, nice=0, itrealvalue=0, starttime=3, vsize=19828736, rss=396, rlim=18446744073709551615, startcode=1, endcode=1, startstack=0, kstkesp=0, kstkeip=0, signal=0, blocked=0, sigignore=4096, sigcatch=536962595, wchan=18446744073709551615, nswap=0, cnswap=0, exit_signal=0, processor=0, rt_priority=0, policy=0, name="init", uid=0, euid=0, gid=0, egid=0, pctcpu=0.0, pctmem=0.08, nlwp=1, cgroup=[], smaps=Sys::ProcTable::Smaps: pid: 1, pss: 0, rss: 0, uss: 0, swap: 0, vss: 0>
```

Note rss has a value coming from /etc/PID/stat but smaps is not readable so we default to all 0's.